File: backend/api/Controllers/DTOs/VillageDetailsDTO.cs
namespace api.Controllers.DTOs;

public class VillageDetailsDTO
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public string? CrestImageUrl { get; set; }
    public List<VillageDetailsMilitaryUnitDTO> MilitaryUnits { get; set; } = null!;
    public List<VillageDetailsBuildingDTO> Buildings { get; set; } = null!;
    public ResourcesDTO AvailableResources { get; set; } = null!;
    public ResourcesDTO ResourcesProductionPerMinute { get; set; } = null!;
    public List<VillageDetailsMilitaryUnitQueueDTO> MilitaryUnitsQueue { get; set; } = null!;
    public List<VillageDetailsBuildingQueueDTO> BuildingsQueue { get; set; } = null!;
}

public class VillageDetailsMilitaryUnitDTO
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public int Amount { get; set; }
    public string? IconUrl { get; set; }
}

public class VillageDetailsBuildingDTO
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public int Level { get; set; }
    public string? ImageUrl { get; set; }
    public int BuildingSpot { get; set; }
}

public class VillageDetailsMilitaryUnitQueueDTO
{
    public Guid Id { get; set; }
    public string MilitaryUnitName { get; set; } = null!;
    public int Amount { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
}

public class VillageDetailsBuildingQueueDTO
{
    public Guid Id { get; set; }
    public string BuildingName { get; set; } = null!;
    public int ToLevel { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
}

File: frontend/src/api/dtos/VillageDetailsDTO.ts
export type VillageDetailsDTO = {
  id: string
  name: string
  crestImageUrl?: string
  militaryUnits: VillageDetailsMilitaryUnitDTO[]
  buildings: VillageDetailsBuildingDTO[]
  availableResources: VillageDetailsResourcesDTO
  resourcesProductionPerMinute: VillageDetailsResourcesDTO
  militaryUnitsQueue: VillageDetailsMilitaryUnitQueueDTO[]
  buildingsQueue: VillageDetailsBuildingQueueDTO[]
}

export type VillageDetailsMilitaryUnitDTO = {
  id: string
  name: string
  amount: number
  iconUrl?: string
}

export type VillageDetailsBuildingDTO = {
  id: string
  name: string
  level: number
  imageUrl?: string
  buildingSpot: number
}

export type VillageDetailsResourcesDTO = {
  wood: number
  iron: number
  wheat: number
  gold: number
}

export type VillageDetailsMilitaryUnitQueueDTO = {
  id: string
  militaryUnitName: string
  amount: number
  startTime: string
  endTime: string
}

export type VillageDetailsBuildingQueueDTO = {
  id: string
  buildingName: string
  toLevel: number
  startTime: string
  endTime: string
}
